pkg/cloudprovider/aws/aws.go

```diff
@@ -97,6 +184,217 @@ func (self *GoamzEC2) GetMetaData(key string) ([]byte, error) {
 	return v, nil
 }
 
+// Implements EC2.DescribeLoadBalancers
+func (self *goamzEC2) DescribeLoadBalancers(region string, findName string) (map[string]elb.LoadBalancer, error) {
+	client, err := self.getElbClient(region)
+	if err != nil {
+		return nil, err
+	}
+
+	request := &elb.DescribeLoadBalancer{}
+	// Names are limited to 32 characters, so we must use tags
+	//request.Names = []string{findName}
+	response, err := client.DescribeLoadBalancers(request)
+	if err != nil {
+		elbError, ok := err.(*elb.Error)
+		if ok && elbError.Code == "LoadBalancerNotFound" {
+			// Not found
+			return nil, nil
+		}
+		glog.Error("error describing load balancers: ", err)
+		return nil, err
+	}
+
+	loadBalancersByAwsId := map[string]elb.LoadBalancer{}
+	for _, loadBalancer := range response.LoadBalancers {
+		awsId := loadBalancer.LoadBalancerName
+		if !strings.HasPrefix(awsId, LOADBALANCER_NAME_PREFIX) {
+			continue
+		}
+
+		// TODO: Cache the name -> tag mapping (it should never change)
+		loadBalancersByAwsId[awsId] = loadBalancer
+	}
+
+	loadBalancersByName := map[string]elb.LoadBalancer{}
+	if len(loadBalancersByAwsId) != 0 {
```

Comment:
This code would be more readable if you changed the condition so that all your code didn't have to be indented inside the if statement:
if len(loadBalancersByAwsId) == 0 {
  return loadBalancerByName, nil
}
// the rest of your logic here